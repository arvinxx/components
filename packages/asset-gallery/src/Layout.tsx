import type { FC, ReactNode } from 'react';
import React from 'react';
import { Columns } from '@bedrock-layout/columns';
import { Slider } from 'antd';

export interface LayoutProps {
  layout?: 'grid' | 'masonry';
  grid?: {
    columns?: number;
    setColumn?: (value: number) => void;
    showSlider?: boolean;
  };
  logo?: ReactNode | string;
}

const Layout: FC<LayoutProps> = ({ children, layout, grid, logo }) => {
  if (layout === 'grid') {
    const { columns, setColumn } = grid;
    return (
      <div>
        <div
          style={{
            padding: '0 8px',
            display: 'flex',
            justifyContent: 'space-between',
            alignItems: 'center',
          }}
        >
          <div>
            {typeof logo === 'string' ? (
              <img src={logo} width={40} alt="logo" />
            ) : (
              logo
            )}
          </div>
          <Slider
            style={{ width: 100 }}
            value={-columns}
            tooltipVisible={false}
            onChange={(v) => setColumn(-v)}
            max={-1}
            min={-4}
          />
        </div>
        <Columns columns={columns}>{children}</Columns>
      </div>
    );
  }

  if (layout === 'masonry')
    return (
      <div className={'avx-asset-gallery-container avx-asset-gallery-masonry'}>
        {children}
      </div>
    );
};

export default Layout;
