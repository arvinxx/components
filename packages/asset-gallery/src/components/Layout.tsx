import type { FC } from 'react';
import React from 'react';
import { Columns } from '@bedrock-layout/columns';
import Header from './Header';

import type { AssetGalleryProps } from '../types';

type LayoutProps = Pick<
  AssetGalleryProps,
  | 'layout'
  | 'showSlider'
  | 'columns'
  | 'logo'
  | 'onColumnsChange'
  | 'onLayoutChange'
  | 'style'
>;

const Layout: FC<LayoutProps> = ({
  onColumnsChange,
  children,
  layout,
  logo,
  columns,
  onLayoutChange,
  showSlider,
  style,
}) => {
  return (
    <div style={style}>
      <Header
        layout={layout}
        onColumnsChange={onColumnsChange}
        columns={columns}
        logo={logo}
        onLayoutChange={onLayoutChange}
        showSlider={showSlider}
      />
      {layout === 'grid' ? (
        <Columns gutter={''} columns={columns}>
          {children}
        </Columns>
      ) : (
        <div
          className={'avx-asset-gallery-container avx-asset-gallery-masonry'}
          style={{ columnCount: columns }}
        >
          {children}
        </div>
      )}
    </div>
  );
};

export default Layout;
