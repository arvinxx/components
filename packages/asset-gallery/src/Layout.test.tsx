import React from 'react';
import { render } from '@testing-library/react';

import Layout from './Layout';

describe('Layout', () => {
  it('瀑布流布局', () => {
    const { container } = render(<Layout layout={'masonry'} />);
    expect(container).toMatchSnapshot();
  });
  // it('Grid 布局', () => {
  //   const { container } = render(
  //     <Layout layout={'grid'} grid={{ columns: 1 }} />,
  //   );
  //   expect(container).toMatchSnapshot();
  // });
});
