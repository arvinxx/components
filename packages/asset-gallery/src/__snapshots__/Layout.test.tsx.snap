// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout 瀑布流布局 1`] = `
<div>
  <div
    class="avx-asset-gallery-container avx-asset-gallery-masonry"
  />
</div>
`;
